src/main/kotlin/com/infomaniak/core/DownloadManager.kt

```diff
@@ -79,7 +79,15 @@ sealed interface DownloadStatus {
 }
 
 suspend fun DownloadManager.startDownloadingFile(request: DownloadManager.Request): UniqueDownloadId? {
-    val uniqueDownloadId = Dispatchers.IO { enqueue(request) }
+    val uniqueDownloadId = Dispatchers.IO {
+        try {
+            enqueue(request)
+        } catch (_: NullPointerException) {
```

Comment:
If the expected behaviour is to receive -1, we should catch all exception with a `runCatching.cancelable()` and `getOrElse`
Others could have implemented it differently and throw other exception

Reply:
Others just respect the doc and return `-1`.

I don't like catching everything as it could hide anything, and we would never see it change. The API "niceness" shouldn't be an enemy for correctness, IMO.

Also, do we want to do defensive programming for all APIs we use just in case a rogue manufacturer decides to pick another victim API?